### ISSUE TITLE
ManagedIdentityCredential throw ClientAuthenticationError instead of CredentialUnavailableError when identity doesn't exist

### ISSUE THREAD

**Issue by evileagle** [Azure/azure-sdk-for-python#10488]
- **azure-identity**: 
- **1.3.0**: 
- **Notebook**:
- **3.6**: 

**Issue**
```
  from azure.identity import (ChainedTokenCredential, ManagedIdentityCredential,InteractiveBrowserCredential, DeviceCodeCredential)
  credentials = [ManagedIdentityCredential()]   
  credentials.append(InteractiveBrowserCredential())
  credentials.append(DeviceCodeCredential(AZURE_CLIENT_ID))
  ChainedTokenCredential(*credentials).get_token(scope)
```
Doesn't work in compute vm that doesn't have managed identity. ChainedTokenCredential directly failed when call ManagedIdentityCredential.get_token, didn't try other methods.

**Expected behavior**
ManagedIdentityCredential should throw CredentialUnavailableError so that ChainedTokenCredential can catch it, and try the next one.

**Screenshots**
![image](https://user-images.githubusercontent.com/5699456/77492792-aa081a80-6dfe-11ea-99fa-da942bc6017d.png)

**Comment by chlowell** [comment]
Thanks for reporting this! I'll fix it for the next release. If you need a workaround in the meantime, one option is to subclass `ChainedTokenCredential` to add a special case to `get_token` ([see this gist](https://gist.github.com/chlowell/44099a29306d5231b0e3038633c039cf) for an example).

**Comment by chlowell** [comment]
This is fixed in [1.3.1](https://pypi.org/project/azure-identity/1.3.1/). Thanks again for opening this issue.
